src/web/View.tsx

```diff
@@ -0,0 +1,116 @@
+import * as React from 'react'
+import * as THREE from 'three'
+import { createPortal, useFrame, useThree } from '@react-three/fiber'
+
+const isOrthographicCamera = (def: any): def is THREE.OrthographicCamera =>
+  def && (def as THREE.OrthographicCamera).isOrthographicCamera
+const col = new THREE.Color()
+
+export type ContainerProps = {
+  scene: THREE.Scene
+  index: number
+  children?: React.ReactNode
+  frames: number
+  rect: React.MutableRefObject<DOMRect>
+  track: React.MutableRefObject<HTMLElement>
+}
+
+export type ViewProps = {
+  /** The tracking element, the view will be cut according to its whereabouts */
+  track: React.MutableRefObject<HTMLElement>
+  /** Views take over the render loop, optional render index (1 by default) */
+  index?: number
+  /** If you know your view is always at the same place set this to 1 to avoid needless getBoundingClientRect overhead */
+  frames?: number
+  /** The scene to render, if you leave this undefined it will render the default scene */
+  children?: React.ReactNode
```

Comment:
Should we not include a `scene` and `camera` prop so that multiple windows can share a scene or camera?

Reply:
sure, would also be possible, you can add your changes to this pr if you want. with the current code it will share the default scene without children, but that arguably doesn't seem very useful and a scene prop would be more versatile.